Services/Authentication/classes/Setup/class.ilSessionMaxIdleIsSetObjective.php

```diff
@@ -0,0 +1,103 @@
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Setup;
+
+class ilSessionMaxIdleIsSetObjective implements Setup\Objective
+{
+    public function __construct(
+        protected ilAuthenticationSetupConfig $config
+    ) {
+    }
+
+    public function getHash(): string
+    {
+        return hash("sha256", self::class);
+    }
+
+    public function getLabel(): string
+    {
+        return "Ensures 'session_max_idle' is set properly";
+    }
+
+    public function isNotable(): bool
+    {
+        return false;
+    }
+
+    public function getPreconditions(Setup\Environment $environment): array
+    {
+        return [
+            new ilDatabaseInitializedObjective(),
+            new \ilSettingsFactoryExistsObjective()
+        ];
+    }
+
+    public function achieve(Setup\Environment $environment): Setup\Environment
+    {
+        $admin_interaction = $environment->getResource(Setup\Environment::RESOURCE_ADMIN_INTERACTION);
+        $settings_factory = $environment->getResource(Setup\Environment::RESOURCE_SETTINGS_FACTORY);
+
+        $session_max_idle = $this->config->getSessionMaxIdle();
+
+        $apache_php_ini_path = $this->getPHPIniPathForApache("localhost/info.php");
+        $apache_php_ini = parse_ini_file($apache_php_ini_path);
+
+        $cookie_lifetime = $apache_php_ini["session.cookie_lifetime"];
+        $gc_maxlifetime = $apache_php_ini["session.gc_maxlifetime"];
+
+        if ($cookie_lifetime != 0) {
+            $message =
+                "The value 'session.cookie_lifetime' in your php.ini does not correspond" . PHP_EOL .
+                "to the value '0' recommended by ILIAS. Do you want to continue anyway?";
+
+            if (!$admin_interaction->confirmOrDeny($message)) {
+                throw new Setup\NoConfirmationException($message);
+            }
+        }
+
+        if ($gc_maxlifetime <= $session_max_idle) {
+            $message =
+                "The value 'session.gc_maxlifetime' in your php.ini is smaller or equal than" . PHP_EOL .
+                "'session_max_idle' in your ILIAS-Config. ILIAS recommends a bigger value." . PHP_EOL .
+                "Do you want to continue anyway?";
+
+            if (!$admin_interaction->confirmOrDeny($message)) {
+                throw new Setup\NoConfirmationException($message);
+            }
+        }
+
+        $settings = $settings_factory->settingsFor("common");
+
+        $settings->set("session_max_idle", (string) $session_max_idle);
```

Comment:
IMO we'll have to set the `expire` value in the `client.ini.php`, not the `session_max_idle` in our `settings` key/value storage.

The `session_max_idle` setting is only used by the "Load-Dependent Session"-mode, which is already removed from the ILIAS 10.x code base (see: https://docu.ilias.de/go/wiki/wpage_8263_1357 / PR #7525 )

Reply:
Done!